python/src/aac/lang/lsp/language_server.py

```diff
@@ -93,29 +94,30 @@ async def did_close(ls: AacLanguageServer, params: DidCloseTextDocumentParams):
     managed_file.is_client_managed = False
 
 
-async def did_change(ls: AacLanguageServer, params: DidChangeTextDocumentParams):
+async def did_save(ls: AacLanguageServer, params: DidSaveTextDocumentParams):
     """Text document did change notification."""
     logging.info(f"Text document altered by LSP client {params.text_document.uri}.")
 
-    file_content = params.content_changes[0].text
+    file_content = params.text
     incoming_definitions = parse(file_content)
```

Comment:
When I test locally and in Gitpod, I get the following error:

```
TypeError: lstat: path should be string, bytes or os.PathLike, not NoneType
```

which turns out to be a result of `params.text` being `None` when the `textDocument/didSave` request is made.

To demonstrate this issue, I opened the `aac/python/model/counter/counter.yaml` file and make a small change to spacing and saved. Again this is happening locally for me and in Gitpod. I think we could use `params.text_document.uri` slightly modified to not include the `file://` part (i.e. something like `params.text_document.uri.removeprefix("file://")`) to resolve what I'm seeing but it seems like this may have already been working for you so I'm open to discussing alternate solutions.

Reply:
Implemented your suggestion. Good catch; Thanks!